llvm/test/compiler-course/suvorov_d_fmsub_optimize_pass/suvorov_d_fmsub_optimize_test.mir

```diff
@@ -0,0 +1,297 @@
+# RUN: llc -mtriple x86_64-unknown-linux-gnu --load=%llvmshlibdir/FmsubOptPass_Suvorov_Dmitrii_FIIT1_BACKEND%shlibext \
+# RUN: -run-pass=fmsub_pass  %s -o - | FileCheck %s
+
+--- |
+  ; ModuleID = 'fmsub_test.ll"
+  target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128"
+  target triple = "x86_64-unknown-linux-gnu"
+  
+  define float @fmsub_ss(float %a, float %b, float %c) #0 {
+    %tmp = fmul float %a, %b
+    %res = fsub float %c, %tmp
+    ret float %res
+  }
+  
+  define double @fmsub_sd(double %a, double %b, double %c) #0 {
+    %tmp = fmul double %a, %b
+    %res = fsub double %c, %tmp
+    ret double %res
+  }
+  
+  define <4 x float> @fmsub_ps(<4 x float> %a, <4 x float> %b, <4 x float> %c) #0 {
+  entry:
+    %tmp = fmul <4 x float> %a, %b
+    %res = fsub <4 x float> %c, %tmp
+    ret <4 x float> %res
+  }
+  
+  define <2 x double> @fmsub_pd(<2 x double> %a, <2 x double> %b, <2 x double> %c) #0 {
+  entry:
+    %tmp = fmul <2 x double> %a, %b
+    %res = fsub <2 x double> %c, %tmp
+    ret <2 x double> %res
+  }
+  
+  attributes #0 = { "target-features"="+fma" }
+
+# CHECK-LABEL: name: fmsub_ss
+# CHECK:       VFNMADD213SSr
```

Comment:
Please, add more complex checks: check that registers and operations sequence are correct